main.go

```diff
@@ -106,6 +106,23 @@ func truncateDate(date time.Time, granularity string) time.Time {
 	}
 }
 
+// Calculates whether or not input data (s3) is more stale than target data (Redshift)
+// Expects:
+// - inputDataDate corresponds to the s3 data timestamp of the job
+// - targetDataDate is the maximum timestamp of the DB table
+// - granularity indicating how often data snapshots are recorded in the target
+func isInputDataOlder(inputDataDate time.Time, targetDataDate *time.Time, granularity string) bool {
```

Comment:
I would change this to isInputDataStale or similar, based on your other comments - I like that naming convention, I think it's way clearer!